lib/pyspark_init.py

```diff
@@ -88,6 +88,14 @@ def load_data(spark: SparkSession) -> DataFrame:
         first("primaryName").alias("primaryName"),
         first("knownForTitles").alias("knownForTitles"),
     )
+    
+    for column in ['primaryProfession', 'knownForTitles']:
+        df_name_basics = df_name_basics.withColumn(column, when(df_name_basics[column] == "\\N", None).otherwise(df_name_basics[column]))
+    for column in ['titleType', 'primaryTitle', 'originalTitle', 'genres']:
+      df_title_basics = df_title_basics.withColumn(column, when(df_title_basics[column] == "\\N", None).otherwise(df_title_basics[column]))
+    for column in ['ordering', 'category', 'job', 'characters']:
+      df_title_principals = df_title_principals.withColumn(column, when(df_title_principals[column] == "\\N", None).otherwise(df_title_principals[column])
```

Comment:
Czy te funkcje nie zmieniają nulli w sourcowych tabelach - df_name_basics, df_title_basics i df_title_principals? Jeśli tak, to nulle nie zostaną zmienione w połączonej tabeli, z której dalej korzystamy, połączenie dzieje się wcześniej. Należy albo zmienić ten kod tak, aby zmieniane były nulle w połączonej tabeli, albo przesunąć ten kod wyżej, żeby nulle były zamienione przed połączeniem tych tabel

Reply:
miał tam być, pomyłka, poprawiłam

Reply:
niestety teraz pojawia się problem z uruchomieniem kodu "local variable 'df_title_basics' referenced before assignment". Prawodopodobnie dlatego, że wcześniej zmienna nazywa się df_title_basic - zmienisz proszę?

Reply:
rzeczywiście, już poprawione

Reply:
Magda, dalej pojawia się błąd. Zmieniłaś nazwę zmiennej w miejscu przypisania nazwy, ale nie wszędzie zmieniłaś nazwę tej zmiennej w dalszej części kodu, więc teraz w innych częściach kodu nie znajduje tej tabeli. Musisz uzgodnić nazwę tej zmiennej w całym pliku.

Reply:
o matko, sorki, powinno być już dobrze.
powód jest taki, że zrobiłam to wyszukując w tekście kodu 'df_title_basic', ale widzę teraz, że przy edytowaniu (tym sposobem nie w colabie) fraza wyszukuje się wtedy tylko we fragmencie, który jest nieprzeskrolowany, nawet po przeskrolowaniu i ponownym przeszukaniu nie znajduje się reszta, więc trzeba np. w colabie